hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-resourcemanager/src/main/java/org/apache/hadoop/yarn/server/resourcemanager/scheduler/fair/FSParentQueue.java

```diff
@@ -305,4 +305,24 @@ public void recoverContainer(Resource clusterResource,
     // TODO Auto-generated method stub
     
   }
+
+  /**
+   * Recursively check policies for queues in pre-order. Get queue policies
+   * from the allocation file instead of properties of {@link FSQueue} objects.
+   *
+   * @param queueConf allocation configuration
+   * @throws AllocationConfigurationException if there is any policy violation
+   */
+  public void checkPoliciesFromConf(AllocationConfiguration queueConf)
```

Comment:
Can this be verifyAndSetPolicyFromConf, and part of FSQueue instead of FSParentQueue. 

In that case, we will not need a separate call to setPolicy.